task-sdk/src/airflow/sdk/definitions/xcom_arg.py

```diff
@@ -355,6 +357,7 @@ def resolve(self, context: Mapping[str, Any]) -> Any:
                 key=self.key,
                 default=NOTSET,
             )
+        result = deserialize(result)
```

Comment:
Why do we need a separate deserialize and "special case" it here.

It doesn't feel right

Reply:
Felt pretty odd to me too, but its cos of https://github.com/apache/airflow/pull/47918/files#diff-62f7d8a52fefdb8e05d4f040c6d3459b4a56fe46976c24f68843dbaeb5a98487R552-R553